### PR TITLE
Fix plugin up, update to 3.0

### PR DESCRIPTION
**Pull request by Ranger-26**
I changed the spawn grenade function to the one in exiled.api, all effects should work fine now

### CODE REVIEW

**Comment by Parkeymon** [comment]
trollage

**Comment by Parkeymon** [comment]
:trollface:

File: LuckyPills/EventHandlers.cs [hunk 1]
```diff
@@ -1,9 +1,10 @@
-﻿namespace LuckyPills
+﻿using InventorySystem;
+
```
Comment (PintTheDragon) [inline LuckyPills/EventHandlers.cs]: Why did you change the styling?

File: LuckyPills/EventHandlers.cs [hunk 2]
```diff
@@ -26,37 +27,8 @@ public EventHandlers(Plugin plugin)
...
-      }
-      else
-      {
-        var relativeVelocity = new Vector3(UnityEngine.Random.Range(0.0f, 2f), UnityEngine.Random.Range(0.0f, 2f), UnityEngine.Random.Range(0.0f, 2f));
-        grenade = UnityEngine.Object.Instantiate(grenadeSettings?.grenadeInstance).GetComponent<Scp018Grenade>();
-        grenade.fuseDuration = timer;
-        grenade.InitData(plyGrenadeManager, relativeVelocity, ply.ReferenceHub.PlayerCameraReference.forward, velocity);
-      }
-      NetworkServer.Spawn(grenade.gameObject);
+      bool fullForce = !(velocity < 1);
```
Comment (PintTheDragon) [inline LuckyPills/EventHandlers.cs]: \>=

**Comment by Ranger-26** [comment]
it works 

> Why did you change the styling redranger?

**Comment by Ranger-26** [comment]
which line?

File: LuckyPills/EventHandlers.cs [hunk 3]
```diff
@@ -130,8 +99,8 @@ private IEnumerator<float> RunPillCoroutine(UsingMedicalItemEventArgs ev)
         {
           var cachedMutatorRole = ev.Player.Role;
           ev.Player.DropItems();
-          ev.Player.SetRole(RoleType.Scp0492, true);
-          Timing.CallDelayed(num, () => ev.Player.SetRole(cachedMutatorRole, true));
+          ev.Player.SetRole(RoleType.Scp0492);
```
Comment (PintTheDragon) [inline LuckyPills/EventHandlers.cs:96]: It was lite for a reason. The zombies will spawn in a black void.

File: LuckyPills/LuckyPills.csproj [hunk 4]
```diff
@@ -33,55 +33,55 @@
     </PropertyGroup>
     <ItemGroup>
         <Reference Include="Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null">
-          <HintPath>..\..\..\Desktop\Exiled\Assembly-CSharp.dll</HintPath>
+          <HintPath>..\..\..\..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\SCP Secret Laboratory Dedicated Server\SCPSL_Data\Managed\Assembly-CSharp-Publicized.dll</HintPath>
```
Comment (PintTheDragon) [inline LuckyPills/LuckyPills.csproj]: Seriously? Like it was bad and you didn't make it worse but this needs to be a relative path.

**Review by PintTheDragon — CHANGES_REQUESTED** [review]

File: LuckyPills/EventHandlers.cs (hunk 1 above)
Comment (Ranger-26) [reply LuckyPills/EventHandlers.cs]: oh rider autocomplete go brrr

File: LuckyPills/EventHandlers.cs (hunk 3 above)
Comment (Ranger-26) [reply LuckyPills/EventHandlers.cs:96]: they do i tested it

File: LuckyPills/LuckyPills.csproj (hunk 4 above)
Comment (Parkeymon) [reply LuckyPills/LuckyPills.csproj]: lma0

File: LuckyPills/Config.cs [hunk 5]
```diff
@@ -1,27 +1,18 @@
...
         public int FlashVomitHealth { get; set; } = 5;
         public float BallVomitInterval { get; set; } = 0.2f;
-
-        public List<string> PossibleEffects { get; set; } = new List<string>()
-        {
-            "mutate",
-            "explode"
-        };
-
+        public List<string> PossibleEffects { get; set; }
```
Comment (PintTheDragon) [inline LuckyPills/Config.cs:15]: Why'd you remove the list?

File: LuckyPills/EventHandlers.cs [hunk 6]
```diff
@@ -27,7 +21,7 @@ public EventHandlers(Plugin plugin)
       float timer,
       float velocity = 1f)
     {
-      bool fullForce = !(velocity < 1);
+      bool fullForce = !(velocity >= 1);
```
Comment (PintTheDragon) [inline LuckyPills/EventHandlers.cs]: You need to remove the ! as well.

File: LuckyPills/Plugin.cs [hunk 7]
```diff
@@ -9,23 +9,23 @@ public class Plugin : Plugin<Config>
         public override string Author { get; } = "Reddking improved by Parkeymon";
         public override string Name { get; } = "Lucky Pills";
         public override Version Version { get; } = new Version(2, 1, 2);
-        public override Version RequiredExiledVersion { get; } = new Version(2, 11, 1, 0);
+        public override Version RequiredExiledVersion { get; } = new Version(3, 0, 0, 0);
```
Comment (Parkeymon) [inline LuckyPills/Plugin.cs:12]: set the required to what you actually use there were breaking changes between 3.0.0 and latest pre-release

**Review by Parkeymon — COMMENTED** [review]
:trollface:

File: LuckyPills/Plugin.cs (hunk 7 above)
Comment (PintTheDragon) [reply LuckyPills/Plugin.cs:12]: The latest EXILED version is listed at 3.0.0.0, no?

Comment (Ranger-26) [reply LuckyPills/Plugin.cs:12]: it doesnt load the plugin if its not 3.0.0.0

File: LuckyPills/Config.cs (hunk 5 above)
Comment (Ranger-26) [reply LuckyPills/Config.cs:15]: cause the original has it like that

**Review by Parkeymon — COMMENTED** [review]
looks correct

File: LuckyPills/EventHandlers.cs [hunk 8]
```diff
@@ -21,7 +21,7 @@ public EventHandlers(Plugin plugin)
       float timer,
       float velocity = 1f)
     {
-      bool fullForce = !(velocity >= 1);
+      bool fullForce = velocity >= 1;
```
Comment (Parkeymon) [inline LuckyPills/EventHandlers.cs:24]: is this right? idk pint look at it

**Review by PintTheDragon — APPROVED** [review]
